src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data.Analysis
+{
+    public enum FieldType
+    {
+        Delimited,
+        FixedWidth
+    }
+
+    public class TextFieldParser : IDisposable
+    {
+        private delegate int ChangeBufferFunction();
+
+        private bool _disposed;
+
+        private TextReader _reader;
+
+        private string[] _commentTokens = new string[0];
```

Comment:
```suggestion
        private string[] _commentTokens = Array.Empty<string>();
```

Reply:
Or just leave it `null` to start.

---
In reply to: [523403969](https://github.com/dotnet/corefxlab/pull/2997#discussion_r523403969) [](ancestors = 523403969)